lib/Cake/Controller/Controller.php

```diff
@@ -389,15 +396,19 @@ public function __isset($name) {
  * @return mixed The requested value for valid variables/aliases else null
  */
 	public function __get($name) {
+		$message = "Property $name is deprecated. Use CakeRequest::$name instead.";
 		switch ($name) {
 			case 'base':
 			case 'here':
 			case 'webroot':
 			case 'data':
+				CakeLog::write(E_USER_DEPRECATED, $message);
```

Comment:
I don't think filling up the logs is going to do any good either. Lot of 2.x plugins / apps might never get updated to remove use of deprecated code.